junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/DisplayNameGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.reflect.Method;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.ClassUtils;
+
+/**
+ * {@code DisplayNameGenerator} defines the {@link Extension} API for
```

Comment:
I'm thinking we should refer to this as an "SPI" instead of an "Extension API" in order to avoid confusion.